src/_P039_Thermocouple.ino

```diff
@@ -350,109 +629,97 @@ float readMax31855(struct EventStruct *event)
 
 float readMax31856(struct EventStruct *event)
 {
-  digitalWrite(Plugin_039_Get_SPI_CS_Pin(event), LOW);
-
-
-  # define P039_RAWVALUE 0
-  # define P039_CR0      1
-  # define P039_CR1      2
-  # define P039_MASK     3
-  # define P039_CJHF     4
-  # define P039_CJLF     5
-  # define P039_LTHFTH   6
-  # define P039_LTHFTL   7
-  # define P039_LTLFTH   8
-  # define P039_LTLFTL   9
-  # define P039_CJTO    10
-  # define P039_CJTH    11
-  # define P039_CJTL    12
-  # define P039_LTCBH   13
-  # define P039_LTCBM   14
-  # define P039_LTCBL   15
-  # define P039_SR      16
+  uint8_t CS_pin_no = get_SPI_CS_Pin(event);
+
 
   uint32_t registers[17] = { 0 };
 
-  for (int i = 0; i < 17; ++i) {
-    registers[i] = SPI.transfer(0x0);
+  for (int i = 0u; i < 17; ++i) {
+    registers[i] = read8BitRegister(CS_pin_no, i);
   }
 
-  // take the SS pin high to de-select the chip:
-  digitalWrite(Plugin_039_Get_SPI_CS_Pin(event), HIGH);
-
-
   uint32_t rawvalue = registers[P039_LTCBH];
   rawvalue = (rawvalue << 8) | registers[P039_LTCBM];
   rawvalue = (rawvalue << 8) | registers[P039_LTCBL];
 
   # ifndef BUILD_NO_DEBUG
 
-  if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
-    String log;
-    log.reserve(66);
-    log = F("P039 : MAX31856 :");
+    if (loglevelActiveFor(LOG_LEVEL_DEBUG)) 
+    {
+      String log;
+      log.reserve(66u);
+      log = F("P039 : MAX31856 :");
 
-    for (int i = 1; i < 17; ++i) {
-      log += ' ';
-      log += String(registers[i], HEX);
+      for (int i = 1; i < 17; ++i) {
+        log += ' ';
```

Comment:
And use can use that `P039_REGISTER_COUNT` here as well

Reply:
Basically "YEPP" - copied.

Reply:
Closed with latest commit and push.